utility_network/display-utility-associations/src/main/java/com/esri/samples/display_utility_associations/DisplayUtilityAssociationsSample.java

```diff
@@ -0,0 +1,245 @@
+        addAssociation();
+        // listen for navigation changes
+        mapView.addNavigationChangedListener((event) -> {
+          addAssociation();
+        });
+      });
+
+      // create ImageView for teo symbol
+      ImageView attachmentImageView = new ImageView();
+      ImageView connectivityImageView = new ImageView();
+      
+      // Create a graphics overlay for associations.
+      associationsOverlay = new GraphicsOverlay();
+      
+      // Create Labels for the preview.
+      Label attachmentLabel = new Label("Attachment");
+      Label connectivityLabel = new Label("Connectivity");
+  
+      // Create Symbols for the associations and preview.
+      SimpleLineSymbol attachmentSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.DOT, 0xFF00FF00, 5);
+      SimpleLineSymbol connectivitySymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.DOT, 0xFFFF0000, 5);
+  
+      // get the symbol from the attachmentSymbol
+      ListenableFuture<Image> attachmentBitmap = attachmentSymbol.createSwatchAsync(0x00000000);
+      attachmentBitmap.addDoneListener(() -> {
+        try {
+          // display the image view in the preview area
+          attachmentImageView.setImage(attachmentBitmap.get());
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided attachmentSymbol" + e.getMessage()).show();
+        }
+      });
+  
+      // get the symbol from the connectivitySymbol
+      ListenableFuture<Image> connectivityBitmap = connectivitySymbol.createSwatchAsync(0x00000000);
+      connectivityBitmap.addDoneListener(() -> {
+        try {
+          // display the image view in the preview area
+          connectivityImageView.setImage(connectivityBitmap.get());
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided connectivitySymbol" + e.getMessage()).show();
+        }
+      });
+      
+      
+      // Create a renderer for the associations.
+      UniqueValueRenderer.UniqueValue attachmentUV = new UniqueValueRenderer.UniqueValue("Attachment", "", attachmentSymbol, Arrays.asList(UtilityAssociationType.ATTACHMENT.toString()));
+      UniqueValueRenderer.UniqueValue connectivityUV = new UniqueValueRenderer.UniqueValue("Connectivity", "", connectivitySymbol, Arrays.asList(UtilityAssociationType.CONNECTIVITY.toString()));
+      associationsOverlay.setRenderer(new UniqueValueRenderer(Arrays.asList("AssociationType"), Arrays.asList(attachmentUV, connectivityUV), "", null));
+      
+      // set initial ArcGISMap extent
+      map.setInitialViewpoint(viewPoint);
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+      mapView.getGraphicsOverlays().add(associationsOverlay);
+      
+      // create a controls box
+      VBox controlsVBox = new VBox();
```

Comment:
I think a Vbox works great here, but it would look neater if it were a grid-pane, and wed arrange the legend to look like so:

| attachment symbol   | Attachment   |
| connectivity symbol | Connectivity |

Reply:
is it okay to have Vbox or do we need to impletment grid-pane  ?